ksqldb-cli/src/main/java/io/confluent/ksql/cli/console/UnclosedQuoteChecker.java

```diff
@@ -0,0 +1,48 @@
+package io.confluent.ksql.cli.console;
+
+/**
+ * Checks to see if the line is in the middle of a quote.  Unlike
+ * org.jline.reader.impl.DefaultParser, this is comment aware.
+ */
+public class UnclosedQuoteChecker {
+  private static final String COMMENT = "--";
+
+  public static boolean isUnclosedQuote(String line) {
+    int quoteStart = -1;
+    for(int i = 0; line != null && i < line.length(); ++i) {
+      if (quoteStart < 0 && isQuoteChar(line, i)) {
+        quoteStart = i;
+      } else if (quoteStart >= 0 && line.charAt(quoteStart) == line.charAt(i) &&
+          !isEscaped(line, i)) {
+        quoteStart = -1;
+      }
+    }
+    int commentInd = line.indexOf(COMMENT);
```

Comment:
This will thrown an NPE if the line is null

Reply:
Maybe add a null check to all methods? `Objects.requireNonNull`

Reply:
Added for the public method.  In general, I'm a little wary of littering these sort of null checks within internal classes, but we seem to do that in a lot of areas in the codebase.